### ISSUE TITLE
run_eagerly model option doesn't work in Keras loss

### ISSUE THREAD

**Issue by Lillypucien** [tensorflow/tensorflow#37870]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Windows 10, x64
- TensorFlow installed from (source or
binary): Binary (pip)
- TensorFlow version (use command below): TF 2.2.0 (2.2.0.dev20200323)
- Python version: 3.7
- CUDA/cuDNN version: 10.1 / 7.6
- GPU model and memory: Bug appears on several computers with different GPU

**Describe the current behavior**

When fitting a Keras model in eager mode, by compiling it with the option run_eagerly=True, the loss is not run eagerly, even though the rest of the model is.

This bug does not appear when using  tf.config.experimental_run_functions_eagerly(True) instead of the run_eagerly option to run the model eagerly, in which case the loss is run eagerly.

**Describe the expected behavior**

Both tf.config.experimental_run_functions_eagerly(True) and run_eagerly=True should have the same behaviour when fitting a model: run all of the model, including its loss, eagerly.

**Standalone code to reproduce the issue** 

```
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers

# Custom loss
class CustomLoss(keras.losses.Loss):
	def call(self, y_true, y_pred):
		print(tf.executing_eagerly())
		
		x = y_true + y_pred
		return tf.reduce_mean(x)

if __name__ == "__main__" :
	data = np.random.random((16, 1000, 3)).astype(np.float32)
	
	inputs = tf.keras.Input(shape=(1000,3))
	outputs = tf.keras.layers.Dense(3)(inputs)
	model = tf.keras.Model(inputs=inputs, outputs=outputs)
	
	# tf.config.experimental_run_functions_eagerly(True) # runs custom loss eagerly
	# model.compile(loss=CustomLoss())
	
	model.compile(loss=CustomLoss(), run_eagerly = True) # does not run custom loss eagerly
	
	model.fit(x=data, y=data)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

[logCompileEagerly.txt](https://github.com/tensorflow/tensorflow/files/4375802/logCompileEagerly.txt)
[logExperimentalRunEagerly.txt](https://github.com/tensorflow/tensorflow/files/4375804/logExperimentalRunEagerly.txt)

**Comment by khimraj** [comment]
@Lillypucien , to run model eagerly first you need to enable eager execution with `tf.enable_eager_execution()`
Complete code is:
```
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
tf.enable_eager_execution()

# Custom loss
class CustomLoss(keras.losses.Loss):
	def call(self, y_true, y_pred):
		print(tf.executing_eagerly())

		x = y_true + y_pred
		return tf.reduce_mean(x)

if __name__ == "__main__" :
	data = np.random.random((16, 1000, 3)).astype(np.float32)
	
	inputs = tf.keras.Input(shape=(1000,3))
	outputs = tf.keras.layers.Dense(3)(inputs)
	model = tf.keras.Model(inputs=inputs, outputs=outputs)
	
	# tf.config.experimental_run_functions_eagerly(True) # runs custom loss eagerly
	# model.compile(loss=CustomLoss())
	
	model.compile(loss=CustomLoss(), run_eagerly = True) # does not run custom loss eagerly
	
	model.fit(x=data, y=data)
```

**Comment by Lillypucien** [comment]
Calling tf.enable_eager_execution() indeed does run all of the model eagerly, including its loss, as described in my post (and shown in logExperimentalRunEagerly.txt). 

However, tf.Keras models also have an attribute run_eagerly which, when set to True, allows a specific model to be run eagerly, whithout interfering with the rest of the code. Setting this attribute to True used to run the whole model eagerly, including the loss (tested with tf version 2.2.0.dev20200304). However, with the last tf nightly version (2.2.0.dev20200323), the loss is no longer run eagerly when run_eagerly is set to True, even though the rest of the model is, as shown in logCompileEagerly.txt.

**Comment by Lillypucien** [comment]
This issue seems to be a direct consequence of issue #37895
